### ISSUE TITLE
Using LSD objective method to evalate the upsample wav, I can't get the same results on the paper. 

### ISSUE THREAD

**Issue by DYJ1111** [kuleshov/audio-super-res#25]
I review the DNN network, using single speaker dataset in vctk, but when i evalate the upsample wav, I can't get the same results. It maybe my LSD code faults or what? 
can you please tell me the split about train and validation datasets?
thank you.

**Comment by JahnKhan** [comment]
iam trying to evaluate it too, at the moment i couldnt get the same results too. It would be nice to have the original model.

**Comment by DYJ1111** [comment]
just run the train code, but can't test the model.

**Comment by JahnKhan** [comment]
how couldnt you test it? 
do you have your own repostitory?

checkout my repository if you want i forked the project.

i run this to validate

` python run.py eval --logname ./full_end_normalized_parmas_ps8.s8-audiounet.lr0.00300.1.g4.b32.d8192.r4.lr0.000300.1.g4.b8/model.ckpt-33 --out-label singlespeaker-out --wav-file-list ../data/vctk/speaker1/speaker1-val-files.txt --r 4`

where `full_end_normalized_parmas_ps8.s8-audiounet.lr0.00300.1.g4.b32.d8192.r4.lr0.000300.1.g4.b8/ `was exported after training.

split and validation as far as i have understood is in those files:

audio-super-res\data\vctk\speaker1\speaker1-train-files.txt"
audio-super-res\data\vctk\speaker1\speaker1-val-files.txt"

**Comment by JahnKhan** [comment]
> just run the train code, but can't test the model.

what code did you use? the main repo from kuleshov? what libraries did you use? i couldnt manage to run it with the libs in the description. I had to change from python 2.X to 3.X and change tensorflow and keras versions.

**Comment by Kolin96** [comment]
Same issue here. My implementation of LSD didn't even get the same scale.

**Comment by JahnKhan** [comment]
> Same issue here. My implementation of LSD didn't even get the same scale, while the SNR's were comparable.

Could you fix your problems ?

**Comment by vinayak19th** [comment]
Did anyone find a fix?

**Comment by vinayak19th** [comment]
``` python
def compute_LSD(x_hr,x_pr):
    def get_power(x):
        S = librosa.stft(x, 2048)
        S = np.log10(np.abs(S)**2 )
        return S    
    with np.errstate(divide='ignore'):
        S1 = get_power(x_hr)
        S2 = get_power(x_pr)
        lsd = np.mean(np.sqrt(np.mean((S1-S2)**2, axis=1)), axis=0)
    return lsd
```

Try using that snippet to calculate LSD. That seemed to help

**Comment by Sawyerb** [comment]
I'm happy to help if this is still an issue. FWIW, I've updated the code to run in Python 3 with later versions of Keras and Tensorflow.
